### PR TITLE
Refactor delete handling to remove the workspace

### LINKED ISSUE #55: Fix for Secret/Lease deletion needs to be reworked
### What happened?
The fix for #40 needs to be reworked since it deletes the Secret/Lease before the reconciliation loop has finished with the Managed Resource.  That causes terraform init errors and leaves the Workspace objects hung.

### What environment did it happen in?
Crossplane version: 1.7

### PR DESCRIPTION
**Pull request by bobh66**
Signed-off-by: Bob Haddleton <bob.haddleton@nokia.com>

### Description of your changes
Refactor the delete handling code to use "terraform workspace delete" to remove the workspace cleanly.

Fixes #55 

I have:

- [X] Read and followed Crossplane's [contribution process].
- [X] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
The code ran in my local environment for a week with no issues.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by bobh66** [comment]
@ytsarev - can you please take a look when you get a chance?  Thanks!

File: internal/terraform/terraform.go [hunk 1]
```diff
@@ -185,6 +187,37 @@ func (h Harness) Workspace(ctx context.Context, name string) error {
...
+	}
+	cmd = exec.CommandContext(ctx, h.Path, "workspace", "delete", "-no-color", name) //nolint:gosec
+	cmd.Dir = h.Dir
+
+	_, err = cmd.Output()
+	if err == nil {
+		// We successfully deleted the workspace; we're done.
+		return nil
+	}
+	// The working directory could be deleted here too instead of waiting for GC to clean it up
```
Comment (ytsarev) [inline internal/terraform/terraform.go:217]: @bobh66 can you extend this comment to `TODO(bobh66): ` so this potential enhancement is greppable?

File: internal/controller/workspace/workspace.go [hunk 2]
```diff
@@ -263,20 +260,23 @@ func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
...
-				return managed.ExternalObservation{}, errors.Wrap(derr, errDiff)
-			}
-			return managed.ExternalObservation{ResourceExists: false}, nil
+		if !meta.WasDeleted(mg) {
+			return managed.ExternalObservation{}, errors.Wrap(err, errDiff)
 		}
-		return managed.ExternalObservation{}, errors.Wrap(err, errDiff)
+		// terraform plan can fail on deleted resources, so let the reconciliation loop
+		// call Delete() if there are still resources in the tfstate file
+		differs = false
```
Comment (ytsarev) [inline internal/controller/workspace/workspace.go:268]: Do we have a test for that logic?

**Review by ytsarev — COMMENTED** [review]
@bobh66 thanks a lot for this contribution, left couple of comments

File: internal/terraform/terraform.go (hunk 1 above)
Comment (bobh66) [reply internal/terraform/terraform.go]: Done

File: internal/controller/workspace/workspace.go (hunk 2 above)
Comment (bobh66) [reply internal/controller/workspace/workspace.go:268]: Added tests for this scenario

**Review by ytsarev — APPROVED** [review]
@bobh66 thanks a lot for the enhancements! LGTM

**Pull request merged**